### DISCUSSION TITLE
This looks like an awesome replacement for zinit

### DISCUSSION THREAD

**Discussion by ryaminal** [marlonrichert/zsh-snap#129] [General]
Hello.

Not certain if this is common knowledge, but last Friday the owner of zinit(and like 15 related sites) decided to just delete everything and leave all of us adopters stranded. There is a great initiative to do zinit-continuum but I'm not thrilled by that initiative.

I'm looking for an alternative and I guess I'm hoping to find a home in zsh-snap.

It appears almost everything is already present in snap, but I'm curious if someone has more information on any discrepancies.

There might be one for installing node and ruby stuff from zinit. In zinit you could do something like: 
```bash
zinit wait lucid for id-as'node-stuff' node'!bash-language-server;' zdharma-continuum/null
```

Anything that's "missing" I'm more than happy to help contribute.

**Comment by marlonrichert** [comment]
> In zinit you could do something like: 
> ```bash
> zinit wait lucid for id-as'node-stuff' node'!bash-language-server;' zdharma-continuum/null
> ```

What is that actually supposed to do? It makes no sense to me.

**Reply by marlonrichert** [reply]
@ryaminal ?

**Reply by squarelover** [reply]
I know this is a bit old, but zinit has sort of its own language. I really don't get this line particularly, but what I liked about zinit is that you can build a config script that all you needed to do was download your zshrc file onto a new machine, and it would automatically install all your tools commands and environment features. I think this is what the OP is requesting.

**Reply by marlonrichert** [reply]
Doesn’t Znap do that, too?

**Reply by chuan137** [reply]
> What is that actually supposed to do? It makes no sense to me.

it manages the user's node_modules/ in a different path and installs the module 'bash-language-server' in it. Zinit supports installing binaries or managing famous tools directly in zshrc

**Reply by marlonrichert** [reply]
I think that's better left to package managers and/or dependency managers.

**Reply by Finkregh** [reply]
Suggestion: try https://mise.jdx.dev/ for these things
